DIPs/DIPxxxx_attributes.md

```diff
@@ -0,0 +1,185 @@
+| DIP:            | (number/id)                                                     |
+| Review Count:   | 0
+| Author:         | Nicholas Wilson                                                 |
+| Implementation: | (links to implementation PR if any)                             |
+| Status:         | Will be set by the DIP manager (e.g. "Approved" or "Rejected")  |
+
+## Abstract
+
+Addresses the desire for different sets of default attributes and rectifies the non-invertibilty
+of the special compiler recognised attributes by having module level defaults. It does not (yet) 
+propose any mechanism to remove compiler attributes directly (e.g. `@!nogc`).
+
+### Links
+
+[Forum discussion](https://forum.dlang.org/thread/wnddmlmfinqqfccdlhqc@forum.dlang.org)
+
+## Terminology
+
+Groups of attributes that are mutually exclusive (such as `@safe`, `@system`, `@trusted`) and presence/absence attributes
+(e.g. `pure`, `nothrow`) and their (currently) non-existant logical negations are called _attribute groups_.
+
+Attributes from `core.attribute` are called core attributes.
+
+## Rationale
+
+Many users feel that the default attributes have the wrong defaults and given that attributes are not invertable,  putting 
+```
+pure: nothrow: @nogc: @safe:
+```
+at the top of a file means that one can never "undo" those attributes.
+This DIP proposes a solution to easily change the default values of the attributes while also allowing symbol specific 
+overrides of individual attribute groups.
+
+## Description
+
+Move all (DMD) compiler recognized attributes into `core.attribute`, making them symbols in their own right, 
+grouping by attribute groups into `enum`s, each with
+* a value `inferred`. The compiler shall determine the value of the attribute. It is illegal to have a function declaration `inferred`.
+* the attribute(s)
+* (if applicable) the attributes' logical negation.
+
+It is suggested that 
+* `inferred` have the value 0, and the compiler shall be required to replace enum core attributes
+that have the value `inferred` to the acual value as determined by the compiler, i.e. no symbols in object files or in reflection shall
+have a core attribute with the value `inferred`. 
+* the current default have the value 1.
+* the negation of the default (`@safe` for `@safe`/`@system`/`@trusted`) the value 2.
+* any other values (e.g. `@trusted`) continue from the value 3.
+
+A module declaration may be tagged with zero or more attribute groups, to apply to all symbols (bar templates which remain inferred with explicit tagging) declared within the module acting as the default.
+If any attribute groups are absent, then the value for that attribute group default to the corresponding value in `core.attribute.defaultAttributeSet`, which will have the values of the current defauls, but may be versioned in druntime as the end user wishes, or with command line switches (e.g. `-safe` or if Type_Info / Module Info generation is added as an attribute `-betterC`).
```

Comment:
Yeah that is hard to read. I think it might be easier to link to the implementation to demonstrate.